run.sh

```diff
@@ -1,19 +1,35 @@
 #!/usr/bin/env bash
 
 startConnIdx=0
+connectorArgsFound=0
 for i in "$@"; do
   startConnIdx=$((startConnIdx+1))
-  if [ $i == "--conf" ]; then break; fi
+  if [ $i == "--conf" ]; then connectorArgsFound=1; break; fi
+  lastParam=$i
 done
 
-if [ $startConnIdx -eq $# ]; then 
+if [ $connectorArgsFound -eq 0 ]; then 
   connectorArgs="" 
-  sparkmaster=""
-  stressArgs=${*:2}
 else 
   connectorArgs=${*:$((startConnIdx))} 
-  sparkmaster="--master ${*:$((startConnIdx-1)):1}"
-  stressArgs=${*:2:$((startConnIdx-2))}
+fi
+
+echo "$lastParam" | grep -E "^(yarn|spark|mesos|local)" >& /dev/null
+noMasterProvided=$? 
+if [ $noMasterProvided -eq 1 ]; then
```

Comment:
Potential minor future improvement: If `$noMasterProvided -eq 1`, then check the list of valid testName's, and if the `$lastParam` is not in those test names, then fail fast.